pkg/synopsysctl/cmd_update.go

```diff
@@ -69,26 +69,27 @@ Update Operator Commands
 */
 
 // getOperatorToUpdate creates a SpecConfig that represents the Synopsys Operator running in the cluster
-func getOperatorToUpdate(namespace string, cmd *cobra.Command) (*soperator.SpecConfig, map[string]string, []string, error) {
+func getOperatorToUpdate(namespace string, cmd *cobra.Command) (*soperator.SpecConfig, map[string]string, []string, bool, error) {
 	crds := []string{}
 	crdMap := make(map[string]string)
-
+	var isClusterScoped bool
 	// check whether the Synopsys Operator config map exist
 	cm, err := util.GetConfigMap(kubeClient, namespace, "synopsys-operator")
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("unable to find the 'synopsy-operator' config map in namespace '%s' due to %+v", namespace, err)
+		return nil, nil, nil, isClusterScoped, fmt.Errorf("unable to find the 'synopsy-operator' config map in namespace '%s' due to %+v", namespace, err)
 	}
 	data := cm.Data["config.json"]
 	var testMap map[string]interface{}
 	err = json.Unmarshal([]byte(data), &testMap)
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("unable to unmarshal config map data due to %+v", err)
+		return nil, nil, nil, isClusterScoped, fmt.Errorf("unable to unmarshal config map data due to %+v", err)
 	}
+
 	if _, ok := testMap["IsClusterScoped"]; ok {
 		configData := &protoform.Config{}
 		err = json.Unmarshal([]byte(data), &configData)
 		if err != nil {
-			return nil, nil, nil, fmt.Errorf("unable to unmarshal config map data due to %+v", err)
+			return nil, nil, nil, isClusterScoped, fmt.Errorf("unable to unmarshal config map data due to %+v", err)
 		}
 		isClusterScoped = configData.IsClusterScoped
```

Comment:
Every error above this line is lying because the function can't/doesn't know if the operator is cluster scoped.  I think it's best to have separate function to determine scope that only returns a scope if it knows what it is or an error